WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/ScanViewModelTest.kt

```diff
@@ -116,8 +127,109 @@ class ScanViewModelTest : BaseUnitTest() {
         verify(fetchScanStateUseCase, times(2)).fetchScanState(site)
     }
 
+    @Test
+    fun `when fix all threats button is clicked, then fix threats confirmation dialog action is triggered`() =
+        test {
+            val observers = init()
+
+            (observers.uiStates.last() as Content).items.filterIsInstance<ActionButtonState>().last().onClick.invoke()
+
+            val fixThreatsDialogAction = observers.navigation.last().peekContent()
+            assertThat(fixThreatsDialogAction).isInstanceOf(OpenFixThreatsConfirmationDialog::class.java)
+        }
+
+    @Test
+    fun `when fix threats confirmation dialog action is triggered, then fix threats confirmation dialog is shown`() =
+        test {
+            val expectedDialogTitle = UiStringRes(R.string.threat_fix_all_warning_title)
+            val expectedDialogMessage = UiStringResWithParams(
+                R.string.threat_fix_all_warning_message,
+                listOf(UiStringText("1"))
+            )
+            val expectedPositiveButtonLabel = R.string.dialog_button_ok
+            val expectedNegativeButtonLabel = R.string.dialog_button_cancel
```

Comment:
1) Minor (🔍): Consider moving the `expected` related variables to the `Assert` part of the test. IMHO the `expected` related variables are not to be confused with the `Arrange` part of any test as that part is only for what happens before the `Act`. Wdyt?
2) Minor (🔍): I would even consider doing the below to avoid some boilerplate:
```
                assertThat(title).isEqualTo(UiStringRes(R.string.threat_fix_all_warning_title))
                assertThat(message).isEqualTo(UiStringResWithParams(
                        R.string.threat_fix_all_warning_message,
                        listOf(UiStringText("1"))
                ))
                assertThat(positiveButtonLabel).isEqualTo(R.string.dialog_button_ok)
                assertThat(negativeButtonLabel).isEqualTo(R.string.dialog_button_cancel)
```

PS: If you agree on point.1 please apply it to the rest of the tests.

Reply:
In 21fb41a